File: integration-test-kotlin/src/test/java/com/andrzejressel/sjs/integrationtestkotlin/AbstractSerializatorTest.java
package com.andrzejressel.sjs.integrationtestkotlin;

import pl.andrzejressel.sjs.serializator.Serializator;

import java.nio.ByteBuffer;

import static org.assertj.core.api.Assertions.assertThat;

abstract class AbstractSerializatorTest<T> {
    protected abstract Serializator<T> getSerializator();

    protected void performTest(T value) {
        var serializator = getSerializator();
        ByteBuffer result = serializator.serializeFlatten(value);
        assertThat(result.hasRemaining()).isFalse();
        assertThat(value).isEqualTo(serializator.deserialize(result.flip()));
    }

}

File: integration-test-kotlin/src/test/kotlin/com/andrzejressel/sjs/integrationtestkotlin/AbstractSerializatorTest.kt
package com.andrzejressel.sjs.integrationtestkotlin

import org.assertj.core.api.Assertions
import pl.andrzejressel.sjs.serializator.Serializator

abstract class AbstractSerializatorTest<T> {
    protected abstract val serializator: Serializator<T>
    protected fun performTest(value: T) {
        val serializator = serializator
        val result = serializator.serializeFlatten(value)
        Assertions.assertThat(result.hasRemaining()).isFalse()
        Assertions.assertThat(value).isEqualTo(serializator.deserialize(result.flip()))
    }
}
